crates/symbolicator-test/src/lib.rs

```diff
@@ -122,65 +124,187 @@ pub fn microsoft_symsrv() -> SourceConfig {
     }))
 }
 
-/// Custom version of warp's sealed `IsReject` trait.
-///
-/// This is required to allow the test [`Server`] to spawn a warp server.
-pub trait IsReject {}
-
-impl IsReject for warp::reject::Rejection {}
-impl IsReject for std::convert::Infallible {}
+/// Gives a [`CommonSourceConfig`] with the given layout type and file types filter.
+pub fn source_config(ty: DirectoryLayoutType, filetypes: Vec<FileType>) -> CommonSourceConfig {
+    CommonSourceConfig {
+        filters: SourceFilters {
+            filetypes,
+            ..Default::default()
+        },
+        layout: DirectoryLayout {
+            ty,
+            ..Default::default()
+        },
+        ..Default::default()
+    }
+}
 
 /// A test server that binds to a random port and serves a web app.
 ///
+/// The server counts all the requests that happen, to be accessed via `accesses` or `all_hits`.
+///
+/// It has a couple of routes with different behavior:
+///
+/// - `/redirect/$path` will redirect to the `$path` url.
+/// - `/delay/$time/$path` will sleep for `$time` and then redirect to `$path`.
+/// - `/msdl/` will redirect to the public microsoft symbol server.
+/// - `/respond_statuscode/$num` responds with the status code given in `$num`.
+/// - `/garbage_data/$data` reponnds back with `$data`.
+/// - `/symbols/` serves the fixtures symbols.
+///
 /// This server requires a `tokio` runtime and is supposed to be run in a `tokio::test`. It
 /// automatically stops serving when dropped.
 #[derive(Debug)]
-pub struct Server {
-    pub handle: tokio::task::JoinHandle<()>,
-    pub socket: SocketAddr,
+pub struct HitCounter {
```

Comment:
Nit: I think it makes more sense to call it `Server` or something along these lines than `HitCounter`, because it's a test server that among other things counts hits.